dev/tasks/docs/github.linux.yml

```diff
@@ -0,0 +1,59 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+{% import 'macros.jinja' as macros with context %}
+{% set pr_number = pr_number|replace("Unset", arrow.branch) %}
+{{ macros.github_header() }}
+
+jobs:
+  test:
+    name: Docs Preview 
+    runs-on: ubuntu-latest
+{{ macros.github_set_env(env) }}
+    steps:
+      {{ macros.github_checkout_arrow(fetch_depth=fetch_depth|default(1))|indent }}
+      {{ macros.github_install_archery()|indent }}
+
+      - name: Execute Docker Build
+        shell: bash
+        env:
+          ARROW_JAVA_SKIP_GIT_PLUGIN: true
+        run: |
+          archery docker run \
+            -e SETUPTOOLS_SCM_PRETEND_VERSION="{{ arrow.no_rc_version }}" \
+            {{ flags|default("") }} \
+            {{ image }} \
+            {{ command|default("") }}
+      - name: Checkout Crossbow
+        uses: actions/checkout@v3
+        with:
+          ref: {{ default_branch|default("master") }}
+          path: crossbow
+          fetch-depth: 1
+      - name: Move docs
+        run: |
+          sudo chown -R ${USER}: build
+          mkdir -p crossbow/docs/pr/{{ pr_number }}
+          rsync -a --delete build/docs/ crossbow/docs/pr/{{ pr_number }}
+      - name: Push changes
+        run: |
+          cd crossbow
+          git config --local user.name "Github Actions"
+          git config --local user.email "github-actions@users.noreply.github.com"
+          git add docs/pr/{{ pr_number }}
+          git commit -m "Add docs preview for PR {{ pr_number }}"
+          git push

```

Comment:
Should we maybe `echo` the URL where the docs are hosted here? Just thinking it would be nice to have a reminder where they appear in the CI output.